lib/blacklight.rb

```diff
@@ -1,23 +1,13 @@
 # frozen_string_literal: true
 
 require 'kaminari'
-require 'blacklight/open_struct_with_hash_access'
-require 'blacklight/nested_open_struct_with_hash_access'
 require 'jbuilder'
+require "zeitwerk"
+loader = Zeitwerk::Loader.for_gem
+loader.ignore("#{__dir__}/generators")
+loader.setup
```

Comment:
I'm surprised a rails engine gem doesn't use zeitwerk already, that we need to specifically load  and start it. But you verified this is necessary?

Reply:
Yes, zeitwerk only works in paths where you tell it to work.

Reply:
I wonder if it's possible to put `config.autoload_paths +=` in `Blacklight::Engine` and let Rails take care of it, instead of setting up our own zeitwerk loader 

I also wonder why we need autoloading of the Blacklight engine files.  Do we need them to be reloaded on-the-fly in dev?  If not, avoiding using zeitwerk/autoloading and simply replacing `autoload` with `require` might suffice?

Reply:
I think the point of using zeitwerk is so that we don't have to require all these files. It'll just read the directory and load them all for us.